packages/astro/src/actions/runtime/virtual/server.ts

```diff
@@ -212,3 +230,80 @@ function unwrapBaseObjectSchema(schema: z.ZodType, unparsedInput: FormData) {
 	}
 	return schema;
 }
+
+/**
+ * Access information about Action requests from middleware.
+ */
+export function getMiddlewareContext(context: APIContext) {
+	const callerInfo = getCallerInfo(context);
+
+	// Prevents action results form being handled on a rewrite.
+	// Also prevents our *own* fallback middleware from running
+	// if the user's middleware has already handled the result.
+	const actionResultAlreadySet = Boolean((context.locals as Locals)._actionPayload);
+
+	let action:
+		| {
+				/** Whether an action was called using an RPC function or by using an HTML form action. */
+				calledFrom: 'rpc' | 'form-action';
```

Comment:
```suggestion
				calledFrom: 'rpc' | 'form';
```

"-action" seems redundant, considering we already expose a field called `action`